src/devices/deviceConnected.js

```diff
@@ -0,0 +1,38 @@
+import Sensor from '../API/sensors/sensor.model';
+import {stream} from '../data-streams/input';
+import Debugger from 'debug';
+
+const debug = Debugger('device-connected');
+
+
+const DEVICE_CONNECTED_EVENT = 'add';
+
+export default function trackDeviceConnection() {
+    stream
+        .filter(message => message.event === DEVICE_CONNECTED_EVENT)
+        .subscribe(saveDevice, onError);
+
+    function saveDevice (data) {
+        Sensor.find({description : data.value.description}, function (error, records) {
```

Comment:
possibly better approach with single query to the db
http://stackoverflow.com/questions/20312163/mongoose-dont-insert-if-element-already-stored
https://docs.mongodb.com/manual/reference/operator/update/setOnInsert/